### ISSUE TITLE
GPU/Parallelism support

### ISSUE THREAD

**Issue by hameerabbasi** [pydata/sparse#379]
According to the paper: https://arxiv.org/abs/2001.00532

**Comment by Mulliken** [comment]
Hi @hameerabbasi ! Do we have any progress on parallelism?

**Comment by hameerabbasi** [comment]
Hi! This will take quite a while; it's a large project. It's on the back burner for now.

If you would like parallelism today, please consider using Dask to wrap PyData/Sparse arrays. That works.

**Comment by Mulliken** [comment]
> Hi! This will take quite a while; it's a large project. It's on the back burner for now.
> 
> If you would like parallelism today, please consider using Dask to wrap PyData/Sparse arrays. That works.

Thank you @hameerabbasi !

**Comment by H4R5H1T-007** [comment]
@hameerabbasi Sir I think [this](https://developer.nvidia.com/blog/unifying-the-cuda-python-ecosystem/?ncid=so-twit-593261#cid=gtcs21_so-twit_en-us) should also help.

**Comment by alexandrupaler** [comment]
I saw the roadmap and it is mentioned that cupy integration will follow. Do you plan a straightforward integration by replacing straightforward API calls? Or do you have another plan in mind? I am asking because cupy integration would be something I am very much looking forward.

**Comment by hameerabbasi** [comment]
@alexandrupaler It isn't as simple as that as we also use Numba to do some algorithmic stuff, and translating that to the GPU is non-trivial.

**Comment by alexandrupaler** [comment]
I understand. The particular method that would be great to have with cupy support is https://github.com/pydata/sparse/blob/d33bf5f4693fbe8bb26c5491a900ea41ac8a2d67/sparse/_common.py#L89 Is at any point numba called behind  the scenes there? The documentation mentions that it is mostly similar to the numpy original.

**Comment by hameerabbasi** [comment]
Yes, the `tensordot` is converted to a 2D dot which is then executed via Numba.

**Comment by alexandrupaler** [comment]
Would using ` NUMBA_DISABLE_JIT` and enabling CuPy through a flag not be a (temporary) solution?
